### ISSUE TITLE
No bookmark found, some bookmarks not created

### ISSUE THREAD

**Issue by fschweitzer-ETHZ** [alphapapa/activities.el#30]
1. Activities was unable to get a buffer for bookmark:

(#("Activity: DC-13	" 0 16 (bmkp-full-record (#1 (time 26058 30868 176769 395000) (visits . 0) (activities-name . "DC-13	") (handler . activities-bookmark-handler)))) (filename . "   - no file -") (buffer-name . "*Ibuffer*") (front-context-string . " MRL Name       ") (rear-context-string) (front-context-region-string) (rear-context-region-string) (visits . 1) (time 26058 31089 378453 71000) (created 26058 30967 111331 208000) (position . 1))

It's likely that the bookmark's file no longer exists, in which case you may need to relocate it and redefine this activity.

If this is not the case, please report this error to the `activities' maintainer.

from the "bookmarks" file -- it is there:
(#1=(#("Activity: DC-13	" 0 16
      (bmkp-full-record #1#))
    (time 26058 30868 176769 395000)
    (visits . 0)
    (activities-name . "DC-13	")
    (handler . activities-bookmark-handler))

2. For other activities, no bookmark was automaticically created

**Comment by alphapapa** [comment]
1. As you can see in that data structure, that bookmark attempts to point to the `*Ibuffer*` buffer.  `ibuffer` does not support the `bookmark` system, so its buffers can't be restored.  I'll need to adjust the error message to help explain that.

2.  I don't understand what you mean.  Please provide a recipe to reproduce the problem.

**Comment by fschweitzer-ETHZ** [comment]
Thanks for the explanation.
Indeed, the activity bookmarks I assumed were missing also referred to 
different *Ibuffers* - that's why they are not there.
But if we cannot use *Ibuffer*, how can "activity" be used to comprise 
the "buffers displayed in a window"? Could you provide an example, or a 
screenshot?

**Comment by alphapapa** [comment]
Have you looked at the documentation?  It should already explain all of that.

**Comment by fschweitzer-ETHZ** [comment]
You are right, and everything works as described (I was probably expecting something like tab-line combined with automatic bookmarking). Thanks for your work.

**Comment by alphapapa** [comment]
You're welcome.

I still need to adjust the error message, so reopening this issue.

**Comment by shackra** [comment]
so is the same with `pdf-view-mode`? I use `pdf-tools` with it:

```
Activities was unable to get a buffer for bookmark:

("original-1830.pdf" (filename . "~/Documentos/Straubinger Digital/oficio-parvo-1830/original-1830.pdf") (position . 1) (page . 96) (slice) (size . fit-page) (origin) (handler . pdf-view-bookmark-jump-handler) (defaults "original-1830.pdf"))

It's likely that the bookmark's file no longer exists, in which case you may need to relocate it and redefine this activity.

If this is not the case, please report this error to the `activities' maintainer.

In the meantime, you may ignore this error and use the other buffers in the activity.

```

**Comment by alphapapa** [comment]
> so is the same with `pdf-view-mode`? I use `pdf-tools` with it:
> 
> ```
> Activities was unable to get a buffer for bookmark:
> 
> ("original-1830.pdf" (filename . "~/Documentos/Straubinger Digital/oficio-parvo-1830/original-1830.pdf") (position . 1) (page . 96) (slice) (size . fit-page) (origin) (handler . pdf-view-bookmark-jump-handler) (defaults "original-1830.pdf"))
> 
> It's likely that the bookmark's file no longer exists, in which case you may need to relocate it and redefine this activity.
> 
> If this is not the case, please report this error to the `activities' maintainer.
> 
> In the meantime, you may ignore this error and use the other buffers in the activity.
> ```

You can see from that bookmark record that it has a `pdf-view-` handler function, so it should work.  Are you sure the file is still there?  If so, try bookmarking just that buffer with `C-x r m` and then kill it and see if you can return to it with `C-x r b`.

**Comment by shackra** [comment]
> > so is the same with `pdf-view-mode`? I use `pdf-tools` with it:
> > ```
> > Activities was unable to get a buffer for bookmark:
> > 
> > ("original-1830.pdf" (filename . "~/Documentos/Straubinger Digital/oficio-parvo-1830/original-1830.pdf") (position . 1) (page . 96) (slice) (size . fit-page) (origin) (handler . pdf-view-bookmark-jump-handler) (defaults "original-1830.pdf"))
> > 
> > It's likely that the bookmark's file no longer exists, in which case you may need to relocate it and redefine this activity.
> > 
> > If this is not the case, please report this error to the `activities' maintainer.
> > 
> > In the meantime, you may ignore this error and use the other buffers in the activity.
> > ```
> 
> You can see from that bookmark record that it has a `pdf-view-` handler function, so it should work. Are you sure the file is still there? If so, try bookmarking just that buffer with `C-x r m` and then kill it and see if you can return to it with `C-x r b`.

Well, I see the file bookmarked

![image](https://github.com/alphapapa/activities.el/assets/1055216/fe479981-67ba-4120-802a-9c6b14ae97e0)

and I was able to come back to it from the bookmarks

**Comment by alphapapa** [comment]
I see an error in your echo area there: `File mode specification error: (void-variable pdf-tools-enabled-modes)`.  That seems likely to be relevant to the problem.

**Comment by shackra** [comment]
> I'm guessing that it indicates a bug in `pdf-tools`'s bookmark support.

if they have that supported at all, of course.

**Comment by alphapapa** [comment]
> > I'm guessing that it indicates a bug in `pdf-tools`'s bookmark support.
> 
> if they have that supported at all, of course.

You can see in the bookmark record that it is, because there's a handler function: `pdf-view-bookmark-jump-handler`
